sdk/core/azure-core/src/main/java/com/azure/android/core/http/responsepaging/AsyncPagedDataCollection.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.http.responsepaging;
+
+import androidx.annotation.NonNull;
+
+import com.azure.android.core.http.Callback;
+import com.azure.android.core.util.paging.Page;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Objects;
+
+import okhttp3.Response;
+
+/**
+ * Represents a collection of pages where page data and response for retrieving it can be returned asynchronously
+ * @param <T> type of the items in the page
+ * @param <P> Page of items
```

Comment:
```suggestion
 * Represents a collection of pages where page data and the response it comes with it can be obtained asynchronously
 * @param <T> type of the items in the page
 * @param <P> type of the page
```